generate/triggergen/main.go

```diff
@@ -302,22 +302,37 @@ package {{.Package}}
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/juju/juju/core/database/schema"
 )
 
 {{range .Views}}
 // ChangeLogTriggersFor{{title .Name}} generates the triggers for the 
 // {{.Name}} table.
-func ChangeLogTriggersFor{{title .Name}}(columnName string, namespaceID int) func() schema.Patch {
+func ChangeLogTriggersFor{{title .Name}}(namespaceID int, changeColumnNames ...string) func() schema.Patch {
+	if len(changeColumnNames) == 0 {
+		panic("must supply the table column to use")
+	}
+	if len(changeColumnNames) > 2 {
+		panic("too many table columns supplied")
+	}
+	changeLogColumns := []string{"changed"}
+	newColumnValues := "NEW." + changeColumnNames[0]
+	oldColumnValues := "OLD." + changeColumnNames[0]
+	if len(changeColumnNames) > 1 {
+		changeLogColumns = append(changeLogColumns, "discriminator")
+		newColumnValues += ", NEW." + changeColumnNames[1]
+		oldColumnValues += ", OLD." + changeColumnNames[1]
+	}
```

Comment:
Don't overload this function. This complicates all trigger code. We can push this into the generation code at _compile_ time and not _runtime_.

Reply:
Done